src/main/java/nurseybook/model/ModelManager.java

```diff
@@ -307,18 +310,15 @@ public boolean equals(Object obj) {
         // state check
         ModelManager other = (ModelManager) obj;
 
-        if (elderlyOfInterest == null) {
-            return other.elderlyOfInterest == null
-                    && versionedNurseyBook.equals(other.versionedNurseyBook)
-                    && userPrefs.equals(other.userPrefs)
-                    && filteredElderlies.equals(other.filteredElderlies)
-                    && filteredTasks.equals(other.filteredTasks);
-        }
-        return other.elderlyOfInterest != null
+        boolean areElderlyOfInterestsEqual = elderlyOfInterest == null && other.elderlyOfInterest == null
+                ? true
+                : elderlyOfInterest == null || other.elderlyOfInterest == null
+                    ? false
+                    : elderlyOfInterest.equals(other.elderlyOfInterest);
+        return areElderlyOfInterestsEqual
```

Comment:
Would it be better if you simply check that `elderlyOfInterest` is null and add an additional line in the `equals()` method for `Elderly` to check if the `other` object is `null`?

Reply:
Ah thanks, was trying to find out a better way for this